support-frontend/assets/helpers/tracking/quantumMetric.ts

```diff
@@ -128,6 +138,78 @@ function sendEventWhenReadyTrigger(sendEventWhenReady: () => void): void {
 	}
 }
 
+function sendEventAcquisitionDataFromQueryParamEvent(
+	acquisitionData: ReferrerAcquisitionData,
+): void {
+	void ifQmPermitted(() => {
+		const sendEventWhenReady = () => {
+			const source = acquisitionData.source;
+			const componentId = acquisitionData.componentId;
+			const componentType = acquisitionData.componentType;
+			const campaignCode = acquisitionData.campaignCode;
+			const referrerUrl = acquisitionData.referrerUrl;
+			const isRemote = acquisitionData.isRemote;
+
+			if (!source) {
+				return;
+			} else if (window.QuantumMetricAPI?.isOn()) {
+				sendEvent(
+					SendEventAcquisitionDataFromQueryParam.Source,
+					false,
+					source.toString(),
+				);
+			}
+			if (!componentId) {
+				return;
+			} else if (window.QuantumMetricAPI?.isOn()) {
+				sendEvent(
+					SendEventAcquisitionDataFromQueryParam.ComponentId,
+					false,
+					componentId.toString(),
+				);
+			}
+			if (!componentType) {
+				return;
+			} else if (window.QuantumMetricAPI?.isOn()) {
+				sendEvent(
+					SendEventAcquisitionDataFromQueryParam.ComponentType,
+					false,
+					componentType.toString(),
+				);
+			}
+			if (!campaignCode) {
+				return;
+			} else if (window.QuantumMetricAPI?.isOn()) {
+				sendEvent(
+					SendEventAcquisitionDataFromQueryParam.CampaignCode,
+					false,
+					campaignCode.toString(),
+				);
+			}
+			if (!referrerUrl) {
+				return;
+			} else if (window.QuantumMetricAPI?.isOn()) {
+				sendEvent(
+					SendEventAcquisitionDataFromQueryParam.ReferrerUrl,
+					false,
+					referrerUrl.toString(),
+				);
+			}
+			if (!isRemote) {
+				return;
+			} else if (window.QuantumMetricAPI?.isOn()) {
+				sendEvent(
+					SendEventAcquisitionDataFromQueryParam.IsRemote,
+					false,
+					isRemote.toString(),
+				);
+			}
+		};
```

Comment:
```suggestion
						if (window.QuantumMetricAPI?.isOn()) {
				const source = acquisitionData.source;
				const componentId = acquisitionData.componentId;
				const componentType = acquisitionData.componentType;
				const campaignCode = acquisitionData.campaignCode;
				const referrerUrl = acquisitionData.referrerUrl;
				const isRemote = acquisitionData.isRemote;

				if (source) {
					sendEvent(
						SendEventAcquisitionDataFromQueryParam.Source,
						false,
						source.toString(),
					);
				}
				if (componentId) {
					sendEvent(
						SendEventAcquisitionDataFromQueryParam.ComponentId,
						false,
						componentId.toString(),
					);
				}
				if (componentType) {
					sendEvent(
						SendEventAcquisitionDataFromQueryParam.ComponentType,
						false,
						componentType.toString(),
					);
				}
				if (campaignCode) {
					sendEvent(
						SendEventAcquisitionDataFromQueryParam.CampaignCode,
						false,
						campaignCode.toString(),
					);
				}
				if (referrerUrl) {
					sendEvent(
						SendEventAcquisitionDataFromQueryParam.ReferrerUrl,
						false,
						referrerUrl.toString(),
					);
				}
				if (isRemote) {
					sendEvent(
						SendEventAcquisitionDataFromQueryParam.IsRemote,
						false,
						isRemote.toString(),
					);
				}
			}
		};
```

Reply:
Just a suggestion , maybe this would this work too with a few less repeated checks maybe.